lib/ContractInteract/BrandedToken.js

```diff
@@ -0,0 +1,382 @@
+'use strict';
+
+const Web3 = require('web3');
+const Contracts = require('../Contracts');
+const AbiBinProvider = require('../AbiBinProvider');
+const Utils = require('../../utils/Utils');
+
+const ContractName = 'BrandedToken';
+
+/**
+ * Contract interact for Branded token.
+ */
+class BrandedToken {
+  /**
+   * Constructor for Anchor.
+   *
+   * @param {Object} web3 Web3 object.
+   * @param {string} address BrandedToken contract address.
+   */
+  constructor(web3, address) {
```

Comment:
In my opinion, the `BrandedToken` class is not concerned with what's origin or auxiliary. It only cares that it is deployed on a chain that it can access with a web3 object.

Reply:
no, rather moving it to `web3` from where it is specified (and ambivalent)